### DISCUSSION TITLE
CLI could not contact EJBCA instance

### DISCUSSION THREAD

**Discussion by rso61** [Keyfactor/ejbca-ce#718] [Q&A] [answered]
I need help on this topic.
I have read all the related threads but I can not derive from one of them.

I have setup Wildfly according to https://docs.keyfactor.com/ejbca/8.3.2/wildfly-26 including 
- Elyton Credential Store
- DB Driver
- Data Source
- Secured https with a cert
- Remoting
- Firewall

I have no error deploying the EAR file with ant deployear
```
[wildfly@deployments]$ ls -l
total 45688
-rw-------. 1 wildfly wildfly 46134433 Nov 20 16:51 ejbca.ear
-rw-r--r--. 1 wildfly wildfly        9 Nov 20 16:51 ejbca.ear.deployed
-rw-r--r--. 1 wildfly wildfly   624101 Nov 20 10:25 mariadb-java-client.jar
-rw-r--r--. 1 wildfly wildfly       23 Nov 20 10:25 mariadb-java-client.jar.deployed
```
I can connect to port 8443 via Browser from other host (getting an Exceptions has occured, but connects)
I tested the DS from the Wildfly Condole => works
No errors in the Wildfly server.log
I can connect to ports 9990, 8443 on localhost using curl.

But still ant runinstall fails with:
```
ejbca:initCA:
     [echo] Initializing CA with 'ManagementCA' 'CN=ManagementCA,O=Everyware AG,C=CH' 'soft' '<ca.tokenpassword hidden>' '4096' 'RSA' '3650' '2.5.29.32.0' 'SHA256WithRSA'   -superadmincn 'SuperAdmin'...
     [java] 11:59:20.298 [main] ERROR org.ejbca.ui.cli.infrastructure.library.CommandLibrary - Error: CLI could not contact EJBCA instance. Either your application server is not up and running, EJBCA has not been deployed successfully, or some firewall rule is blocking the CLI from the application server.
     [java]
     [java] Please be aware that most commands will not work without the application server available.
     [java]
     [java] Exception in thread "main" javax.ejb.NoSuchEJBException: EJBCLIENT000079: Unable to discover destination for request for EJB StatelessEJBLocator for "ejbca/cesecore-ejb/GlobalConfigurationSessionBean", view is interface org.cesecore.configuration.GlobalConfigurationSessionRemote, affinity is None
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:620)
```

thanks for your help!

**Comment by primetomas** [comment]
there are two things:
1. check the [server.log](https://docs.keyfactor.com/ejbca/latest/troubleshooting-guide) of wildfly
2. if you configured remoting correctly, you'll see that remoting uses a different port, a number that was part of the remoting setup commands. That has to work.

**Reply by rso61** [reply]
Ports 8080 and 9990 are configured for wildfly and 8443 for Webapps.
I found one additional port configured 4447 (http://XX.XX.XX.XX:4447/ejbca/ejbcaws/ejbcaws) in the log, As of remoting. But I cannot connect using http (connection refused). But this might be normal for WebService.

Beside of this I found nothing to worry about. 
But I am not very familiar with wildfly/jboss, so I have attached the log file.

[server.log](https://github.com/user-attachments/files/17916575/server.log)

--added--
does ant runinstall connect to this port?

**Reply by rso61** [reply]
**ant runinstall** should connect to which port?
using: **tcpdump -i any port XXXX** I can not see any traffic (where XXXX is any of 8080 8443 9090 3306 or 4447)
what am I doing wrong?

**Reply by mike-agrenius-kushner** [reply]
Hi, 

ANT uses the CLI to create the initial Management CA, which in turn uses Remote EJB, which by default runs via port 4447. You can verify in jboss-ejb-client.properties to check the port used.

Cheers,
Mike

**Reply by rso61** [reply]
I installed everything from scratch. somewhere was an setup/config error. 
Now I am getting a response from port 4447 using http.

But also now I am getting an authentication error
```
     [java]     Suppressed: javax.security.sasl.SaslException: Authentication failed: all available authentication mechanisms failed:
     [java]    JBOSS-LOCAL-USER: javax.security.sasl.SaslException: ELY05128: Failed to read challenge file [Caused by java.io.FileNotFoundException: /opt/wildfly/standalone/tmp/auth/local10433943249297720963.challenge (Permission denied)]
```
what credentials is the CLI expecting?
in which property file are the defined?
do the need to be configured using /opt/wildfly/bin/add-user.sh?

**Reply by mike-agrenius-kushner** [reply, marked as answer]
Shouldn't be, but go over how you set up Elysium in Wildfly...or could it just be a rights issue in the Wildfly directory?

**Reply by rso61** [reply]
ant runinstall needs a lot of permissions to read and write in the wildfly directory structure, This may be mentioned in the installation procedure.

**Reply by primetomas** [reply]
Does it? Like what? runinstrall does not change any wildfly configuration or copy files. 
"ant deployear" copies ejbca.ear to the deployment directory, but otherwise?
